src/app/core/llm/fake-streaming.provider.ts

```diff
@@ -0,0 +1,101 @@
+import { faker } from '@faker-js/faker';
+import { LLMProvider, ProviderInfo, CreateChatParams } from './types';
+
+export interface FakeProviderConfig {
+  avgDelayMs?: number; // average delay between chunks
+  jitterMs?: number; // +/- jitter
+  sentences?: [number, number]; // min/max sentences per reply
+  tokensPerChunk?: [number, number]; // min/max chars per chunk
+  initialThinkingMs?: number; // initial delay before first chunk
+}
+
+export class FakeStreamingProvider implements LLMProvider {
+  info: ProviderInfo = {
+    id: 'fake',
+    label: 'Fake Provider (Debug)',
+    supportsStreaming: true,
+    needsBYOK: false,
+  };
+
+  constructor(private cfg: FakeProviderConfig = {}) {}
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  setKey(_key: string | null): void {
+    // Fake provider doesn't need a key
+  }
+
+  async createChat({
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    messages: _messages,
+    stream,
+    onToken,
+    abortSignal,
+  }: CreateChatParams): Promise<{ text: string } | void> {
+    const sentencesCount = this.randBetween(
```

Comment:
Please label instead of using magic numbers `1` and `3` on lines 35 and 36. Or if you're trying to get a random number, use Faker to generate a random number.